packages/core/src/common/audioAlertPlayer.ts

```diff
@@ -0,0 +1,112 @@
+import { Howl } from 'howler';
+import { leastSevere, mostSevere, defaultAudioSrc } from './constants';
+
+export class AudioAlertPlayer {
+  private _isMuted = false;
+  private _isPlaying = false;
+  private _maxVolume = 1.0;
+  private _severity: number | undefined;
+  private _soundID: number | undefined;
+  private _player: Howl | undefined;
+
+  public isPlaying(): boolean {
+    return this._isPlaying;
+  }
+
+  public isMuted(): boolean {
+    return this._isMuted;
+  }
+
+  public mute(): void {
+    if (this._player && this._soundID) {
+      this._player.mute(true, this._soundID);
+    }
+    this._isMuted = true;
+  }
+
+  public unmute(): void {
+    if (this._player && this._soundID) {
+      this._player?.mute(false, this._soundID);
+    }
+    this._isMuted = false;
+  }
+
+  /* plays the incoming alert if no other alert is playing, or if the incoming alert has
+     a higher severity than the currently playing one. Returns true if incoming alert is
+     played, false otherwise */
+  public play = ({ severity, volume, audioSrc }: { severity: number; volume: number; audioSrc?: string }): boolean => {
+    severity = Math.max(Math.min(severity, leastSevere), mostSevere);
+    // if there's another alert playing, stop if new alert is more severe
+    if (this.isMoreSevere(severity)) {
+      this.stop();
+    }
+
+    volume = Math.max(Math.min(volume, 1.0), 0.0);
+    if (!this.isPlaying()) {
+      const newPlayer = new Howl({
+        src: [audioSrc ?? defaultAudioSrc[severity - 1]],
+        volume: volume * this.getMaxVolume(),
+        onend: () => {
+          // updates AudioAlertPlayer when sound ends
+          this.stop();
+        },
+      });
+
+      this._soundID = newPlayer.play();
+      this._player = newPlayer;
+      this._severity = severity;
+      this._isPlaying = true;
+      if (this.isMuted()) {
```

Comment:
Does it make sense to retain the "muted" state across different instances of alerts? Also I notice you call play() prior to muting - will this play the sound, or will it always mute first due to the order of execution of synchronous code / promise queue?